ir/function.h

```diff
@@ -1,12 +1,102 @@
 #ifndef FUNCTION_H
 #define FUNCTION_H
 
+#include <unordered_map>
+#include <string>
+#include <variant>
+
 #include "inst.h"
 
+class Class;
+class Layer;
+
 class Function {
 public:
-    std::vector<Inst> insts;
-    std::vector<VRegType> vregs;
+    Function(std::string name) : name_(name) {}
+    void AddInst(Inst* inst)
+    {
+        inst->SetFunction(this);
+        for (auto input: inst->GetInputs()) {
+            assert(input != nullptr);
```

Comment:
Consider replacing `assert` with custom macros `ASSERT` (see `macros.h`) so it could be removed in `RELEASE` build type

Reply:
Actually, standard assert from <cassert> is also removed in release build. But for consistency can replace it with ASSERT